src/js/common/browserInfoHelper.js

```diff
@@ -38,7 +38,7 @@ export const getDeviceLocalIPAsString = () => {
         deviceIpString = ip.join(",");
         resolve(deviceIpString);
       }
-      if (event.candidate.candidate) {
+      else if (event.candidate.candidate) {
```

Comment:
I think I know what was bugging me about this fix.
The code inside this `else if` block will only be executed in the following scenario:

`!event` or `!event.candidate` evaluates to true (i.e. event or event.candidate evaluate to false, e.g. are undefined or null)
AND THEN
`event.candidate.candidate` evaluates to true

It seems to me that this would be impossible.

I think what the first `if` block is doing is an example of the bouncer pattern, i.e. doing an early check and if something is wrong, exiting early. So it would be wrong to put this code inside an else block.

Here was my previous suggested fix: https://github.com/intuit/user-data-for-fraud-prevention/pull/3/files

Having thought about it a bit more maybe the fix here is to check on line 33 like this:

`if (!event || !event.candidate || !event.candidate.candidate)`

And if we have that, then we can actually remove the if check on line 41, since logically we would have already resolved earlier in the code.

What do you think?

Reply:
Are you saying the code inside the if block on line 33 has to be executed even if event.candidate.candidate is undefined or null?

Reply:
As things stand with your change, the code inside the block on line 43 will only execute if `event` or `event.candidate` evaluate to `false` AND `event.candidate.candidate` evaluates to `true`. This seems like a logical impossibility to me and essentially the code inside the block on line 43 will never execute.